feat(DatePicker): Add ability to customise format

thyhjwb6 opened the pull request:
## Related issue
Closes #1399

## Overview
Added a `format` prop allows the consumer to supply a custom format based on [Unicode Technical Standard #35](https://www.unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table)

## Reason
Applications need consistent formatting for their context.

## Work carried out
- [x] Expose ability to customise date format

m7kvqbe1 reviewed (changes requested):
LGTM - I guess it's kind of already covered by existing tests.

Just a quick comment about the docs-site change.

thyhjwb6 committed:
feat(DocsSite): Allow HTML to be rendered

Within the docs we would like to render links.

thyhjwb6 committed:
feat(DatePicker): Add ability to customise format

A `format` prop allows the consumer to supply a
custom format based on:
Unicode Technical Standard #35
